### DISCUSSION TITLE
No way to enabled captcha in 3.x

### DISCUSSION THREAD

**Discussion by corneliusio** [solspace/craft-freeform#612] [Questions & Answers] [answered]
We have recpatcha setup for a site still running Craft and Freeform 3.x, however, the captchas do not display on the front-end and there is no option to enable captchas on a form in the dashboard. 

Captchas are enabled in Captcha Settings and have tested with three different sets of Google Recaptcha keys. We have a site running Craft and Freeform 4.x and the option to enable captchas shows up as expected. Captcha Settings are identical for both of these sites and we have tested rendering the form with absolutely no other markup, styles, or javascript on the page—still no recaptcha.

<img width="349" alt="Screenshot 2023-02-28 at 2 33 12 PM" src="https://user-images.githubusercontent.com/4140885/221974295-e84c6d84-c90d-452d-8da5-551dafbeb03e.png">
<img width="345" alt="Screenshot 2023-02-28 at 2 33 03 PM" src="https://user-images.githubusercontent.com/4140885/221974297-640e22a6-b09c-4819-a0a8-d6454dc16462.png">

**Craft & Plugin Info (please complete the following information):**

- Craft Version: 3.7.67
- Freeform Version: 3.13.24
- Freeform Edition: Pro
- Fresh Install or Upgrade: Upgrade

**Comment by kjmartens** [comment]
Hi @corneliusio,

Is there any chance that the 3.x site is using reCAPTCHA v2 Checkbox instead? If so, this is because you just need to remove the special reCAPTCHA field type from the field layout to disable it from that form, whereas the other options are invisible and need a setting. 🙂

**Reply by corneliusio** [reply]
Unfortunately no—I already double-checked this. Set up new reCAPTCHA v3 and reCAPTCHA v2 Checkbox keys and tested each with the appropriate selection in the plugin settings for each.

**Reply by kjmartens** [reply]
Would you be able to send a screenshot of the Captcha settings page in Freeform (and blur out or remove the keys from the inputs for security)? And can you send a screenshot of the entire form layout inside the form builder? 🙂

**Reply by corneliusio** [reply]
No problem. I had to comp a few screen shots for the form layout, so let me know if you need the sidebar configuration too.

<img width="1515" alt="Screenshot 2023-02-28 at 4 04 40 PM" src="https://user-images.githubusercontent.com/4140885/221993247-499b9cbb-5eb7-4bd0-bb16-30cb1d885b48.png">

<img width="908" alt="Screenshot 2023-02-28 at 4 07 25 PM" src="https://user-images.githubusercontent.com/4140885/221993254-eb4d040a-ace4-4cc3-9971-d059ea3eac5a.png">

**Reply by corneliusio** [reply]
I also just discovered another factor here which may actually mean I need to do some more digging on my end, but it looks like this is only occurring when Stripe is enabled on the form. When I removed the Stripe payment field in the screenshot above the recaptha appeared as expected.

**Comment by kjmartens** [comment, marked as answer]
Ok, this makes sense now. When using Stripe Payments with Freeform, it automatically disables reCAPTCHA for that form, so the setting to enable/disable gets hidden as well. I believe reCAPTCHA being disabled automatically was due to Stripe and reCAPTCHA conflicting when attempting to validate the form.

**Comment by corneliusio** [comment]
Is this reCAPTCHA specific or would it apply to hCaptcha as well? Our client has been dealing with an increase in spam submissions and many of their forms integrate Stripe payments. I can understand how it seems odd to need to add spam protection to a form that theoretically charges the end user, but I believe many of these payments are conditional based on other submitted data so the forms can still get spam without ever needing to create a charge.

**Reply by kjmartens** [reply]
I'm not sure how this could be, though. Freeform will only process the form if a proper credit card is being used. Or are you saying that phony/fake credit card numbers are used and you just end up receiving these anyway? 🙂

**Reply by corneliusio** [reply]
Honestly, I may be way off base there. Trying to account for edge cases and presumed that the credit card field could be conditional based on another field value, but it's been a hot minute since I interacted with Freeform's conditional logic. I'll need to go back to the client at this point and get some more details on what they're seeing, but it does look like I have enough info here to take back to them regarding their issues. The reality is if the Stripe field can't be conditional then I suspect this is actually an issue with spam submissions getting past the recaptcha itself on forms that don't have a Stripe integration. 

Just to confirm, this limitation is true for both reCAPTCHA **and** hCaptcha, yes? And, while I understand it seems questionably necessary, are there any plans to address this conflict between captcha and Stripe down the road?

**Reply by kjmartens** [reply]
Currently it's not possible to use conditional logic on a Payment-enabled form. The form will expect to fully validate with Stripe before allowing it to be submitted.

In the future, we plan to find a solution for the conflict issue, and also plan to allow conditional logic on payment processing. 🙂

**Reply by corneliusio** [reply]
Awesome, thanks for the info and for helping get to the bottom of what was happening on our end (again). 😬
